Plugin/ConfigSectionPaymentPlugin.php

```diff
@@ -24,25 +42,65 @@ class ConfigSectionPaymentPlugin
 
     /**
      * @param \Omise\Payment\Model\Config\Config $config
+     * @param OmiseHelper $helper
+     * @param Magento\Framework\Message\ManagerInterface $messageManager
      */
-    public function __construct(Config $config)
-    {
+    public function __construct(
+        Config $config,
+        OmiseHelper $helper,
+        ManagerInterface $messageManager
+    ) {
         $this->config = $config;
+        $this->helper = $helper;
+        $this->messageManager = $messageManager;
+        // using same version as omise-php 2.13(2017-11-02)
+        define('OMISE_API_VERSION', '2017-11-02');
     }
 
     /**
      * @param \Magento\Config\Model\Config $coreConfig
      */
     public function beforeSave(CoreConfig $coreConfig)
     {
-        if ('payment' === $coreConfig->getData('section')) {
-            $keys = $this->getKeys($coreConfig->toArray());
+        if ('payment' === $coreConfig->getSection()) {
 
+            $this->config->setStoreId($coreConfig->getData('store'));
+            $omiseConfigData = $coreConfig->toArray()['groups']['omise'];
+            $keys = $this->getKeys($omiseConfigData);
+            
             // if both keys are empty then we ignore the check.
             if ($keys['public_key'] || $keys['secret_key']) {
                 try {
                     // Fetching capabilities to check the supplied keys validity
-                    OmiseCapabilities::retrieve($keys['public_key'], $keys['secret_key']);
+                    $this->capabilities = OmiseCapabilities::retrieve($keys['public_key'], $keys['secret_key']);
+
+                    /** when using test mode is will fetch all available payment methods
+                     *  that omise is supported
+                     * */
+                    $paymentList  = $this->getBackends();
+                    $omiseConfigPaymentList=$this->getActivePaymentMethods($omiseConfigData);
+
+                    // list active payment methods that not support from capabilities api
+                    $nonSupportPayments = [];
+
+                        // set disable only not support payment methods
+                    $data = $coreConfig->getGroups();
+                    foreach ($omiseConfigPaymentList as $payment => $title) {
+                        if (! in_array($payment, $paymentList)) {
+                            $data['omise']['groups'][$payment]['fields']['active']['value'] = 0;
+                            array_push($nonSupportPayments, $title);
+                        }
+                    }
+
+                        // show error message by using title from omise helper
+                    if (! empty($nonSupportPayments)) {
+                        $this->messageManager->
+                            addError(__("This Omise account is not support " . implode(", ", $nonSupportPayments)));
+                    }
```

Comment:
Great job with the work. I have one suggestion. Let's abstract this code and create a separate function responsible for create a data array. It will be cleaner.

On line 101, comment marked off by indentation.

Reply:
yeah that good idea!! 👍  don't know why `Eclipse` is format comment like that 👀  -> moved https://github.com/omise/omise-magento/pull/361/commits/b4453793481fd9f26ce71050b649697485e51880 🙏